draft-irtf-cfrg-hpke.md

```diff
@@ -892,6 +886,39 @@ security models and assumptions, and do not consider attackers capable of quantu
 computation. A full proof of post-quantum security would need to take this
 difference into account, in addition to simply using a post-quantum KEM.
 
+## Pre-Shared Key Recommendations {#security-psk}
+
+In mode PSK and mode AuthPSK, the PSK SHOULD be of length Nh bytes or
+longer, and SHOULD have Nh bytes of entropy or more. Using a PSK shorter
+than Nh bytes is permitted. A PSK that is longer than Nh bytes or that
+has more than Nh bytes of entropy, respectively, does not increase the
+security level of HPKE, because the extraction step involving the PSK
+only outputs Nh bytes.
+
+HPKE is specified to use HKDF as key derivation function. HKDF is not
+designed to slow down dictionary attacks, see {{?RFC5869}}. Thus,
+HPKE's PSK mechanism is not suitable for use with a low-entropy
+password as the PSK: in scenarios in which the adversary knows the
+KEM shared secret zz, it can perform a dictionary attack on the PSK.
+The adversary can use different oracles to decide if a guess was good,
+depending on the scenario. First, on observation of an HPKE ciphertext,
```

Comment:
In this scenario, the adversary didn't guess the PSK value, so I don't think it's a relevant example.

Reply:
The adversary can try decryption for each PSK value it wants to guess. This is an example of a passive adversary. Would replacing “on observation of an HPKE ciphertext” by “with a captured HPKE ciphertext” make this clearer?

Reply:
If the adversary doesn't have access to the ephemeral shares, this isn't possible, right?